package-examples/ghost/statefulset-mariadb.yaml

```diff
@@ -21,15 +21,9 @@ spec:
           securityContext:
             runAsNonRoot: true
             runAsUser: 1001
-          env:
-            - name: BITNAMI_DEBUG
-              value: "true"
-            - name: MARIADB_USER
-              value: "bn_ghost"
-            - name: MARIADB_DATABASE
-              value: "bitnami_ghost"
-            - name: ALLOW_EMPTY_PASSWORD
-              value: "true"
+          envFrom:
+            - configMapRef:
+                name: ghost-env
```

Comment:
One disadvantage of using the same configmap here is that any config change to that configmap affects restarting ghost-app and mysql. ghost-app config is superset of mysql. Like changing the ghost-app's host is going to restart the mysql for now reason.

So an optimal solution could be, define two config-maps. one for db config and another for pure ghost-app config.

ghost-app's config will be provided by both the config-map and mysql one will be powered by just the db configmap.

Reply:
> One disadvantage of using the same configmap here is that any config change to that configmap affects restarting ghost-app and mysql. ghost-app config is superset of mysql. Like changing the ghost-app's host is going to restart the mysql for now reason.
> 

That's true. Restarting both to align the env change is expected behavior. One issue I had today is that I only updated the env config for MariaDB (so we saw the MariaDB pod works) but not Ghost host. 

> So an optimal solution could be, define two config-maps. one for db config and another for pure ghost-app config.
> 
> ghost-app's config will be provided by both the config-map and mysql one will be powered by just the db configmap.

Sure, I can use two ConfigMaps since the `envFrom` accept a list

Reply:
Actually thought some more and realized, this looks nice from config organization perspective but functionally is inferior to previous approach of `inline config` in the deployment itself. Because now I have to worry about updating the deployment or statefulset otherwise new change will not rollout and in the previous approach it will happen automatically :)

Reply:
Another thing I discovered is that the `envFrom` [cannot be updated](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L2355). This seems not to be the model we are looking for.